src/main/java/hudson/plugins/plot/MatrixPlotPublisher.java

```diff
@@ -99,7 +99,7 @@ public void addPlot(Plot plot) {
             List<Plot> list = groupMap.get(urlGroup);
             list.add(plot);
         } else {
-            List<Plot> list = new ArrayList<Plot>();
+            List<Plot> list = new ArrayList<>();
```

Comment:
@svanoort PTAL
![image](https://user-images.githubusercontent.com/3036347/30195220-e39a8590-940c-11e7-9c18-de733fe9df9b.png)
and 
![image](https://user-images.githubusercontent.com/3036347/30195228-ef9876a4-940c-11e7-8ee8-7e38f1110e61.png)

Reply:
Yeah, that's perfectly valid for Java 7 level, but the latest core is on Java 8, and we're gradually switching things over to compile to Java 8 level, which for complex reasons doesn't always support the diamond syntax (See: https://stackoverflow.com/questions/27134626/when-does-diamond-syntax-not-work-in-java-8). 

I wouldn't necessarily change this back (not that big a deal), but it's worth leaving explicit types in place in the future -- the IDE is just offering a style suggestion but you don't *need* to remove the explicit type with diamond syntax (it won't cause any form of error).  Plus having the type in there makes it easier when this plugin switches to Java 8 support in the future.

Reply:
I hope that explains a bit better?  It's  legit to do this, but I'm suggesting not to do it more, because it'll make work easier for you in the future.